internal/resource/http.go

```diff
@@ -70,6 +80,103 @@ func (f *Fetcher) UpdateHttpTimeouts(timeouts types.Timeouts) {
 
 	f.client.transport.ResponseHeaderTimeout = time.Duration(responseHeader) * time.Second
 	f.client.client.Transport = f.client.transport
+
+	// Update CAs
+	if len(cas) == 0 {
+		return nil
+	}
+
+	pool, err := x509.SystemCertPool()
+	if err != nil {
+		f.Logger.Err("Unable to read system certificate pool: %s", err)
+		return err
+	}
+
+	for _, ca := range cas {
+		u, err := url.Parse(ca.Source)
+		if err != nil {
+			f.Logger.Err("Unable to parse CA URL: %s", err)
+			return err
+		}
+
+		cablob, err := f.getCABlob(u, ca)
+		if err != nil {
+			return err
+		}
+		block, _ := pem.Decode(cablob)
+		if block == nil {
+			f.Logger.Err("Unable to decode CA (%s)", u)
+			return ErrPEMDecodeFailed
+		}
+
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			f.Logger.Err("Unable to parse CA (%s): %s", u, err)
+			return err
+		}
+
+		f.Logger.Info("Adding %q to list of CAs", cert.Subject.CommonName)
+		pool.AddCert(cert)
+	}
+
+	f.client.transport.TLSClientConfig = &tls.Config{RootCAs: pool}
+	f.client.client.Transport = f.client.transport
+	return nil
+}
+
+func (f *Fetcher) getCABlob(u *url.URL, ca types.CaReference) ([]byte, error) {
+	if blob, ok := f.client.cas[ca]; ok {
+		return blob, nil
+	}
+	hasher, err := util.GetHasher(ca.Verification)
+	if err != nil {
+		f.Logger.Crit("Unable to get hasher: %s", err)
+		return nil, err
+	}
+
+	var expectedSum []byte
+	if hasher != nil {
+		// explicitly ignoring the error here because the config should already
+		// be validated by this point
+		_, expectedSumString, _ := ca.Verification.HashParts()
+		expectedSum, err = hex.DecodeString(expectedSumString)
+		if err != nil {
+			f.Logger.Crit("Error parsing verification string %q: %v", expectedSumString, err)
+			return nil, err
+		}
+	}
+
+	cablob, err := f.FetchToBuffer(*u, FetchOptions{
```

Comment:
Not really a comment for this review, but we should probably refactor `f.FetchToBuffer` to take a `types.Verification`